File: common/common.vault/common.vault.model/src/main/java/org/eclipse/slm/common/vault/model/LoginRequest.java
package org.eclipse.slm.common.vault.model;

public class LoginRequest {
    public String role = "default";
    public String jwt;

    public LoginRequest() {
    }

    public LoginRequest(String jwt) {
        this.jwt = jwt;
    }

    public LoginRequest(String role, String jwt) {
        this.role = role;
        this.jwt = jwt;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public String getJwt() {
        return jwt;
    }

    public void setJwt(String jwt) {
        this.jwt = jwt;
    }
}

File: common/common.consul/common.consul.model/src/main/kotlin/org/eclipse/slm/common/consul/model/authentication/LoginRequest.kt
package org.eclipse.slm.common.consul.model.authentication

class LoginRequest {
    var authMethod = "keycloak"
    var bearerToken: String? = null

    constructor() {}

    constructor(bearerToken: String?) {
        this.bearerToken = bearerToken
    }

    constructor(authMethod: String, bearerToken: String?) {
        this.authMethod = authMethod
        this.bearerToken = bearerToken
    }
}
